### PR TITLE
fix(root-listeners): apply listen-on-root mixin to other components

### LINKED ISSUE #683: [dropdown] menu closes and re-opens after click on item
I have the following dropdown (in a larger template, but I dont think that matters):
```
<b-nav-item-dropdown text="Tags">
  <b-dropdown-item to="/tag">Tags</b-dropdown-item>
</b-nav-item-dropdown>
```

When I click on the dropdown-item I expect the dropdown menu to close and stay closed, but actually it only closes for a brief moment and then opens again. See below for the event output, am I missing something or is this an unwanted feature?

image

### PR DESCRIPTION
**Pull request by alexsasharegan**
Closes #683

### CODE REVIEW

**Comment by alexsasharegan** [comment]
Weird, I don't know why I'm showing changes in the docs.

**Comment by tmorehouse** [comment]
THat happend to me once.  Did you do a squash merge from the master?  IF so, it looses context of which other commits have happened.

**Comment by tmorehouse** [comment]
maybe try a merge from master branch back into your branch?

File: lib/mixins/form.js [hunk 1]
```diff
@@ -1,17 +1,34 @@
 export default {
+    computed: {
+        inputClass() {
+            return [
+                this.size ? `form-control-${this.size}` : null,
+                this.state ? `form-control-${this.state}` : null
+            ];
+        },
+        custom() {
```
Comment (tmorehouse) [inline lib/mixins/form.js]: Hmm, this shouldn't  be in here anymore... s plain was moved only to the form-* components that have that option.

Same for the inputClass() computed prop (doesn't apply to all form components)

**Comment by alexsasharegan** [comment]
Yeah, I attempted a squash merge a while back so my PRs would stop showing a laundry list of merge commits, but it not worky muy bien.

File: lib/mixins/form.js [hunk 2]
```diff
@@ -1,17 +1,34 @@
...
-        required: {
+        plain: {
             type: Boolean,
             default: false
+        },
+        state: {
+            type: String
+        },
+        size: {
+            type: String
```
Comment (tmorehouse) [inline lib/mixins/form.js]: And state and size should not be in form.js any more either, as they have been moved over to the components that only need it.

**Comment by alexsasharegan** [comment]
We're down to the readme changes just being editor clean-up on save.

File: docs/components/table/README.md [hunk 3]
```diff
@@ -1,569 +1,43 @@
 # Tables
 
-> For displaying tabular data. `<b-table>` supports pagination, filtering, sorting,
-custom rendering, events, and asynchronous data.
+> For tabular data. Tables support pagination and custom rendering.
```
Comment (tmorehouse) [inline docs/components/table/README.md]: THhs looks like a revert of the new table features?

File: lib/utils/base-adapter.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
+export default class BaseAdapter {
```
Comment (tmorehouse) [inline lib/utils/base-adapter.js]: Mybe these adapter additions should be moved into a separate PR?

**Comment by alexsasharegan** [comment]
@tmorehouse, reload the PR and check the diff again for changes. I think I've fixed this.

File: lib/components/table.vue [hunk 5]
```diff
@@ -75,7 +75,8 @@
 
 <script>
     import { warn } from '../utils';
-    import { keys } from '../utils/object.js'
+    import { keys } from '../utils/object.js';
+    import { listenOnRootMixin } from '../mixins'
```
Comment (tmorehouse) [inline lib/components/table.vue]: semicolon? :stuck_out_tongue_winking_eye:

Comment (alexsasharegan) [inline lib/components/table.vue]: hahaha

Comment (alexsasharegan) [inline lib/components/table.vue]: Goodness, look at me. I tried to fight my editor prefs and add a semicolon, and then I forget it in the very next line!

Comment (tmorehouse) [inline lib/components/table.vue]: Haha!

**Pull request merged**